src/transformers/models/mamba/modeling_mamba.py

```diff
@@ -243,13 +288,12 @@ def slow_forward(self, input_states, cache_params: Optional[MambaCache]=None):
 
         # 2. Convolution sequence transformation
         if cache_params is not None:
-            ssm_state = cache_params.ssm_states[self.layer_idx].clone()
+            # ssm_state = cache_params.ssm_states[self.layer_idx].clone()
+            ssm_state = cache_params.get_ssm_sate(self.layer_idx).clone()
             ssm_state = ssm_state.to(hidden_states.device)
-            if cache_params.seqlen_offset > 0:
-                conv_state = cache_params.conv_states[self.layer_idx]                   # [batch, intermediate_size, conv_kernel_size]
-                conv_state = torch.roll(conv_state, shifts=-1, dims=-1)
-                conv_state[:, :, -1] = hidden_states[:, :, 0]
-                cache_params.conv_states[self.layer_idx].copy_(conv_state)
```

Comment:
this is most probably not gonna be backward compatible. But ready to do so if perf are a lot better!

Reply:
can you please elaborate more on why it's not backward compatible?

Reply:
sorry, since we keep the cache, and update it , it should be BC.